### DISCUSSION TITLE
Install&run plugNmeet server, client and recorder in windows machine

### DISCUSSION THREAD

**Discussion by genciT5** [mynaparrot/plugNmeet-server#500] [Q&A]
Hello there, 

I just found out about this amazing project, and I was planning to use it in my personal project. 

I'm coding in Windows 11 OS. 
I was going through the installation process and faced many issues. Tried to go through the docs, but since it is showing the process only for Linux OS, it is a bit difficult to follow it.
I was wondering whether you have something for windows users or plan to do so? 

Ps. I am planning to develop a c# nuget-package library for plugNMeet in the near-future, and open-source it for everyone to be able to use it.

**Comment by jibon57** [comment]
Sorry, I do not have experience to use windows machine.

**Comment by ravalosv** [comment]
I'm trying to install it on Windows, and although I still can't get it to work completely, I leave you with a guide to give you an idea. I hope it helps you.
*****
To install PlugNMeet on Docker, follow these steps:

1.- Install Docker

2.- Clone the PlugNMeet repository:
Open a terminal and clone the PlugNMeet repository from GitHub.
git clone https://github.com/mynaparrot/plugNmeet-server.git
cd plugNmeet-server

2.1.- Inside the plugNmeet-server folder rename the following files:
     docker-compose_sample.yaml => docker-compose.yaml
     config_sample.yaml => config.yaml
     ingress_sample.yaml => ingress.yaml
     livekit_sample.yaml => livekit.yaml
 
3.- Build and run Docker containers:
docker-compose build

4.- Once built, lift the containers.
docker-compose up -d

5.- Check that everything is working:
docker-compose ps

6.- You can also review the logs to make sure that there are no errors.
docker-compose logs -f

7.- Download the PlugNMeet client 
https://github.com/mynaparrot/plugNmeet-client/releases/download/v1.5.18/client.zip

8.- Unzip the zip and put it in the client/dist path

8.1.- Inside the client/dist/assets folder rename the file
     config-sample.js => config.js

9.- That's all! You can now run the app on the route: 
     http://localhost:8080/login.html

By the way, the data for "plugNmeet API Key" and "plugNmeet API Secret" can be found in the config.yaml file of the server.

**Reply by deenodj** [reply]
> [plugnmeet-api internal] load metadata for docker.io/library/golang:1.23:
time="2024-12-11T16:09:21+01:00" level=warning msg="project has been loaded without an explicit name from a symlink. Using name \"plugNmeet-server\""       
[+] Running 6/6
 ✘ livekit Error         context canceled                                                                                                              1.3s 
 ✘ redis Error           context canceled                                                                                                              1.3s 
 ✘ db Error              failed to resolve reference "docker.io/library/mariadb:10.11": failed to authorize: failed to fetch oauth token:...           1.3s 
 ✘ livekit-ingress Error context canceled                                                                                                              1.3s 
 ✘ nats Error            context canceled                                                                                                              1.3s 
 ✘ etherpad Error        context canceled                                                                                                              1.3s 
Error response from daemon: failed to resolve reference "docker.io/library/mariadb:10.11": failed to authorize: failed to fetch oauth token: unexpected stat
us from GET request to https://auth.docker.io/token?scope=repository%3Alibrary%2Fmariadb%3Apull&service=registry.docker.io: 401 Unauthorized
PS C:\Users\marcel.PALWMD\plugNmeet-server> docker-compose logs -f
time="2024-12-11T16:12:23+01:00" level=warning msg="project has been loaded without an explicit name from a symlink. Using name \"plugNmeet-server\""

**Reply by jibon57** [reply]
Looks like you have connection problem with docker.

**Reply by deenodj** [reply]
Yes thanks it was, when i log off the docker and try it again than its works fine, thank you.
Do you know where i can find my files of plugnmeet localy from my docker?

other problems comes up where i post it here.
https://github.com/mynaparrot/plugNmeet-server/discussions/361#discussioncomment-11536689

Thanks alot, good work
